### ISSUE TITLE
Error and some docs mention strict-arith instead of strict_arith

### ISSUE THREAD

**Issue by Crestwave** [oilshell/oil#507]
Not sure if this is intended. Also, is it explained anywhere why this is enabled by default?

**Comment by andychu** [comment]
I updated the docs.

I turned it on by default a few releases ago, I guess because I wanted there to be some "carrot" to use OSH.  Like if you run it, it will find a problem automatically.

My thought was to turn it off if anyone complains, but so far nobody has.  I can't think of a situation where you wouldn't want to fix the errors it give, but I'd be interested in counterexamples.

I guess there is a much bigger carrot now in the Oil language.

**Comment by Crestwave** [comment]
The error (and apparently a comment) still hasn't been updated (https://github.com/oilshell/oil/search?utf8=✓&q=strict-arith&type=).

I keep getting bitten by it; as usual, it's used in neofetch and pfetch, and a couple of my scripts. And I'm not exactly sure what I'm supposed to do instead.

In my Brainfuck interpreter, I do `(( tape[ptr] = tape[ptr]+1 & 255 ))` to increment the cell, which allows it to extend dynamically in Bash. I could use `${tape[ptr]:-0}`, but then it could get interpreted as an operator (it can't get set to a string in this program, but still). And what if I used `(( ++tape[ptr] ))`?

Another use-case is for using a variable as a boolean, false by default. For example: https://github.com/Crestwave/haikuporter/blob/04fcee2ddf94573a7d0e524084f1f78056582aae/tools/cargo-to-recipe.sh#L5. I could set it to 0 outside of the function, but that seems ugly, and I would think that OSH would be advocating for local variables.

**Comment by andychu** [comment]
OK, I changed it back, since you can always opt into it.  (You could have opted out with `shopt -u strict_arith 2>/dev/null || true`, but I think that's the wrong default.)


I didn't understand all your use cases, but it sounds like there are a bunch of them.

I can see that `++counter` and `++cell[i]` without initialization are probably relatively common idioms... although it will still warn about that, since it always has.

Do you think I should get rid of the warnings too?

Personally, I always like to initialize variables to zero, but I can see that there are a lot of scripts out there that won't.

**Comment by andychu** [comment]
This is what it looks like now... there are a couple other warnings in the codebase and I've been meaning to turn them into hard errors or silent.  I am not sure the concept of "warnings" is that useful.

```
$ bin/osh -c '(( x++ )); echo x=$x'
  (( x++ )); echo x=$x
     ^
[ -c flag ]:1: warning: Undefined value in arithmetic context
x=1
```

**Comment by andychu** [comment]
Released with 0.7.pre6 : https://www.oilshell.org/release/0.7.pre6/
